### PR TITLE
Fix positions on new projects and status suggestions having a blank edition field

### PR DESCRIPTION
**Pull request by MaartenS11**
When testing the frontend after we added the edition security annotation, I noticed that the fronted could not fetch project positions, this because the position fields were set to the default value and not the same value as the project edition. This PR fixes that by mapping over the positions from the `projectRegistration` and creating a new position with the edition field set correctly.

### CODE REVIEW

**Comment by Mikxox** [comment]
I don't think the population script sets the edition field for the positions anyway. I also don't pass the edition for the positions from the frontend post / patch project, should I be doing that as well? Since the patch endpoint can also create new positions for a project.

**Comment by MaartenS11** [comment]
> I don't think the population script sets the edition field for the positions anyway. I also don't pass the edition for the positions from the frontend post / patch project, should I be doing that as well? Since the patch endpoint can also create new positions for a project.

I was just going to mention the patch endpoint. This pr makes it so positions are created with the edition properly set when creating a project but that behaviour doesn't really work when patching so we could either change the frontend to just include the editions or try and make the backend force the editions to the edition from the path.

**Comment by Mikxox** [comment]
Fixing the status suggestions frontend would be adding the same thing on line 82, in frontend/components/student/studentview.tsx
Again this would also need to be fixed in the population script in that case.

**Comment by MaartenS11** [comment]
I ended up forcing the edition fields on the backend, I think this is a better solution than having edition fields everywhere on the frontend when the url already indicates the edition.

File: backend/src/main/kotlin/be/osoc/team1/backend/entities/Project.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Position(
     val amount: Int,
     @JsonIgnore
     @NotBlank
-    val edition: String = ""
+    var edition: String = ""
```
Comment (TomAlard) [inline backend/src/main/kotlin/be/osoc/team1/backend/entities/Project.kt:35]: Why did you change this?

**Review by Mikxox — APPROVED** [review]

File: backend/src/main/kotlin/be/osoc/team1/backend/entities/Project.kt (hunk 1 above)
Comment (MaartenS11) [reply backend/src/main/kotlin/be/osoc/team1/backend/entities/Project.kt:35]: This was changed because there is no other way to patch a project and have the edition fields be set on the positions without requiring the frontend to provide them. You can't make new positions because that would give them new id's so you have to change the existing edition field.

**Review by TomAlard — APPROVED** [review]
